agent/app/agent_test.go

```diff
@@ -74,12 +79,44 @@ func setup(t *testing.T) (*gomock.Controller,
 		mock_factory.NewMockSaveableOption(ctrl)
 }
 
+func TestDoStartMinimumSupportedDockerVersionTerminal(t *testing.T) {
+	ctrl, credentialsManager, state, imageManager, client,
+		dockerClient, stateManagerFactory, saveableOptionFactory := setup(t)
+	defer ctrl.Finish()
+
+	oldAPIVersions := []dockerclient.DockerVersion{
+		dockerclient.Version_1_18,
+		dockerclient.Version_1_19,
+		dockerclient.Version_1_20}
+	gomock.InOrder(
+		dockerClient.EXPECT().SupportedVersions().Return(oldAPIVersions),
+	)
+
+	cfg := getTestConfig()
+	cfg.Checkpoint = true
+	ctx, cancel := context.WithCancel(context.TODO())
+	// Cancel the context to cancel async routines
+	defer cancel()
+	agent := &ecsAgent{
+		ctx:                   ctx,
+		cfg:                   &cfg,
+		dockerClient:          dockerClient,
+		stateManagerFactory:   stateManagerFactory,
+		saveableOptionFactory: saveableOptionFactory,
+	}
+
+	exitCode := agent.doStart(eventstream.NewEventStream("events", ctx),
+		credentialsManager, state, imageManager, client)
+	assert.Equal(t, exitcodes.ExitTerminal, exitCode)
```

Comment:
Could you please add a test case for `exitcodes.ExitError` as well?

Reply:
added TestDoStartMinimumSupportedDockerVersionError test case